src/sympc/tensor/static.py

```diff
@@ -119,4 +121,170 @@ def cat_share_tensor(session_uuid_str: str, *shares: Tuple[ShareTensor]) -> Shar
+    pairwise_comparisons = x_pairwise >= 0
+
+    # re-compute row_length
+    _dim = -1 if dim is None else dim
+    row_length = x.shape[_dim] if x.shape[_dim] > 1 else 2
+
+    result = pairwise_comparisons.sum(0)
+    result = result >= (row_length - 1)
+    res_shape = res_shape[1:]  # Remove the leading dimension because of sum(0)
+
+    if not one_hot:
+        if dim is None:
+            check = result * torch.Tensor([i for i in range(np.prod(res_shape))])
+        else:
+            size = [1 for _ in range(len(res_shape))]
+            size[dim] = res_shape[dim]
+            check = result * torch.Tensor([i for i in range(res_shape[_dim])]).view(
+                size
+            )
+
+        if dim is not None:
+            argmax = check.sum(dim=dim, keepdim=keepdim)
+        else:
+            argmax = check.sum()
+            if (argmax >= row_length).reconstruct():
+                # In case we have 2 max values, rather then returning an invalid index
+                # we raise an exception
+                raise ValueError("There are multiple argmax values")
+
+        result = argmax
+
+    return result
+
+
+def argmax(
+    x: MPCTensor,
+    dim: Optional[Union[int, Tuple[int]]] = None,
+    keepdim=False,
+) -> MPCTensor:
+    """Compute argmax using pairwise comparisons. Makes the number of rounds fixed, here it is 2.
+
+    This is inspired from CrypTen.
+
+    Args:
+        x (MPCTensor): the MPCTensor that argmax will be computed on
+        dim (Union[int, Tuple[int]): compute argmax over a specific dimension(s)
+        keepdim (bool): when one_hot is true and dim is set, keep all the dimensions of the tensor
+
+    Returns:
+        The the index of the maximum value as an MPCTensor
+    """
+    return helper_argmax(x, dim=dim, keepdim=keepdim, one_hot=False)
+
+
+def max_mpc(
+    x: MPCTensor,
+    dim: Optional[Union[int, Tuple[int]]] = None,
+    keepdim=False,
+) -> Union[MPCTensor, Tuple[MPCTensor, MPCTensor]]:
+    """Compute the maximum value for an MPCTensor.
+
+    Args:
+        x (MPCTensor): MPCTensor to be computed the maximum value on.
+        dim (Optional[Union[int, Tuple[int]]]): The dimension over which to compute the maximum.
+        keepdim (bool): when one_hot is true and dim is set, keep all the dimensions of the tensor
+
+    Returns:
+        A tuple representing (max MPCTensor, indices_max MPCTensor)
+    """
+    argmax_mpc = helper_argmax(x, dim=dim, keepdim=keepdim, one_hot=True)
+    max_mpc = argmax_mpc * x
+    if dim is None:
+        res = max_mpc.sum()
+    else:
+        shape = argmax_mpc.shape
+        size = [1 for _ in range(len(shape))]
+        size[dim] = shape[dim]
+        argmax_mpc = argmax_mpc * torch.Tensor([i for i in range(shape[dim])]).view(
+            size
+        )
+
+        max_mpc = max_mpc.sum(dim=dim, keepdim=keepdim)
+        res = max_mpc, argmax_mpc.sum(dim=dim, keepdim=keepdim)
+    return res
+
+
+# from syft < 0.3.0
```

Comment:
### Part 4
Maybe: This can be a part of the docstring like how CrypTen is acknowledged in the `def argmax`